diff --git a/dist/runtime/transformers/path-meta.js b/dist/runtime/transformers/path-meta.js
index 4104e5c976d3c5a88f2ea87427d42a4b31174d61..af4b5867a73cf570514f08b0f3445aed3a2ab0c5 100644
--- a/dist/runtime/transformers/path-meta.js
+++ b/dist/runtime/transformers/path-meta.js
@@ -1,6 +1,6 @@
 import { pascalCase } from "scule";
 import slugify from "slugify";
-import { withoutTrailingSlash, withLeadingSlash } from "ufo";
+import { withLeadingSlash, withoutTrailingSlash } from "ufo";
 import { defineTransformer } from "./utils.js";
 const SEMVER_REGEX = /^(\d+)(\.\d+)*(\.x)?$/;
 export const describeId = (id) => {
@@ -28,7 +28,7 @@ export default defineTransformer({
     const _locale = locales.includes(parts[0]) ? parts.shift() : defaultLocale;
     const filePath = generatePath(parts.join("/"), { respectPathCase });
     return {
-      _path: filePath,
+      _path: content._path ?? filePath,
       _dir: filePath.split("/").slice(-2)[0],
       _draft: content._draft ?? isDraft(_path),
       _partial: isPartial(_path),
