diff --git a/dist/runtime/server/storage.mjs b/dist/runtime/server/storage.mjs
index 698bfc82cdc0bd668a74364eae2b1c382db34084..d68569cd5326868077ad1e1e5b602cb4ed7f67a8 100644
--- a/dist/runtime/server/storage.mjs
+++ b/dist/runtime/server/storage.mjs
@@ -149,6 +149,14 @@ export function serverQueryContent(event, query, ...pathParts) {
         params.where.push({ _locale: contentConfig.defaultLocale });
       }
     }
+    if (params.where) {
+      params.where = params.where.map((w) => {
+        if (typeof w._path === 'string') {
+          w._path = decodeURI(w._path)
+        }
+        return w
+      })
+    }
     return params;
   };
   return queryBuilder;